src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1264,6 +1262,7 @@ public Status do_POST_Document_revs_diff(Database _db, String _docID, String _at
 
         // Look them up, removing the existing ones from revs:
         try {
+            // query db only, not necessary to be syncrhonized
```

Comment:
@hideki I took my word back. Android SQLiteDatabase seems to be thread-safe between read/write regardless of connection pool size. The reason behind this is related to the default number of connection pool size. By default the connection pool size is one and when enabling WAL mode by API not SQL command, connection pool size is 2 or 4 (I don't remember exactly).

```
WAL provides more concurrency as readers do not block writers and a writer does not block readers. Reading and writing can proceed concurrently.
```

We could chat more about what we should do :)